File: adventofcode/year2022/Day01.cs
using adventofcode.common;

namespace adventofcode.year2022;

public class Day01 : Solution
{
    private string[] _input;
    private List<int> _calories;
    
    public Day01(string year, string day) : base(year, day)
    {
        _input = this.LoadInputAsLines();
        _calories = new List<int>();
    }
        
    public override string PartOne()
    {
        var largestCalorie = 0;
        var currentCalorie = 0;
        foreach (var calorie in _input)
        {
            if (string.IsNullOrEmpty(calorie))
            {
                // blank reached, check if calories is new largest
                if (currentCalorie >= largestCalorie)
                {
                    largestCalorie = currentCalorie;
                }
                _calories.Add(currentCalorie);
                currentCalorie = 0;
            }
            else
            {
                currentCalorie += Convert.ToInt32(calorie);
            }
        }

        return Convert.ToString(largestCalorie);
    }

    public override string PartTwo()
    {
        return Convert.ToString(_calories.OrderByDescending(x => x).Take(3).Sum());
    }
}

File: adventofcode/year2019/day01.py
from __future__ import annotations
from adventofcode.common import Solution


class Day01(Solution):
    def __init__(self, year: str, day: str):
        super().__init__(year, day)
        self._input = [int(l) for l in self._load_input_as_lines()]

    def part_one(self):
        return sum(module // 3 - 2 for module in self._input)

    def part_two(self):
        total_fuel = 0
        for module in self._input:
            fuel = module // 3 - 2
            while fuel > 0:
                total_fuel += fuel
                fuel = max(0, fuel // 3 - 2)

        return total_fuel
